### PR TITLE
More examples using HMatrix

### LINKED ISSUE #1213: The legend of the graphics in MetaModelValidation is wrong
The following script produces a polynomial chaos on the cantilever beam example.

```
# Chaos on cantilever beam
import openturns as ot

dist_E = ot.Beta(0.9, 3.1, 2.8e7, 4.8e7)
F_para = ot.LogNormalMuSigma(3.0e4, 9.0e3, 15.0e3) # in N
dist_F = ot.ParametrizedDistribution(F_para)
dist_L = ot.Uniform(250., 260.) # in cm
dist_I = ot.Beta(2.5, 4, 310., 450.) # in cm^4

myDistribution = ot.ComposedDistribution([dist_E, dist_F, dist_L, dist_I])

dim_input = 4 # dimension of the input
dim_output = 1 # dimension of the output

def function_beam(X):
    E, F, L, I = X
    Y = F* (L**3) /  (3 * E * I)
    return [Y]

g = ot.PythonFunction( dim_input, dim_output, function_beam)
g.setInputDescription(myDistribution.getDescription())

# ## Create a polynomial chaos decomposition
multivariateBasis = ot.OrthogonalProductPolynomialFactory([dist_E, dist_F, dist_L, dist_I])

# Generate an training sample 
inputTrain = myDistribution.getSample(50)
outputTrain = g(inputTrain)

# Chaos
selectionAlgorithm = ot.LeastSquaresMetaModelSelectionFactory()
projectionStrategy = ot.LeastSquaresStrategy(inputTrain, outputTrain, selectionAlgorithm)
totalDegree = 5
enumfunc = multivariateBasis.getEnumerateFunction()
P = enumfunc.getStrataCumulatedCardinal(totalDegree)
adaptiveStrategy = ot.FixedStrategy(multivariateBasis, P)

chaosalgo = ot.FunctionalChaosAlgorithm(inputTrain, outputTrain, myDistribution, adaptiveStrategy, projectionStrategy)
chaosalgo.run()

# Get the result
result = chaosalgo.getResult() # get the result
metamodel = result.getMetaModel() # get the metamodel

# ## Validate the metamodel
# Generate a validation sample (which is independent of the training sample).
inputTest = myDistribution.getSample(100)
outputTest = g(inputTest)

val = ot.MetaModelValidation(inputTest, outputTest, metamodel)

# Plot the observed versus the predicted outputs.
import openturns.viewer as otv
graph = val.drawValidation()
otv.View(graph)
```
Metamodel-validation-bug

The legend of this graphics is obviously wrong: the blue line is *not* the model and the red points are *not* the metamodel. This legend must be simply removed. 

Below is a fix, along with an enhancement which prints the Q2 coefficient in the title (the Q2 is available within the validation object):
```
graph = val.drawValidation()
Q2 = val.computePredictivityFactor()
graph.setTitle("Q2=%.2f%%" % (Q2*100))
graph.setLegends(["",""])
otv.View(graph)
```

Metamodel-validation-fixed

My suggestion is that this should be the default setting.

### PR DESCRIPTION
**Pull request by sofianehaddad**
The purpose here is to illustrate the usage of HMatrix in OT.

The applications are :
  - Kriging 
 - Gaussian process
 - \+ minor fixes (bad legends, new methods to Kriging to easily set the "HMatrix" method )

It closes #1213 & #1222

### CODE REVIEW

File: lib/src/Uncertainty/Algorithm/MetaModel/Kriging/KrigingAlgorithm.cxx [hunk 1]
```diff
@@ -258,6 +258,20 @@ Point KrigingAlgorithm::getNoise() const
   return glmAlgo_.getNoise();
 }
 
+String KrigingAlgorithm::getMethod() const
+{
+  UnsignedInteger method = glmAlgo_.getMethod();
```
Comment (regislebrun) [inline lib/src/Uncertainty/Algorithm/MetaModel/Kriging/KrigingAlgorithm.cxx]: const

File: python/src/KrigingAlgorithm_doc.i.in [hunk 2]
```diff
@@ -296,3 +296,22 @@ Returns
...
+// ---------------------------------------------------------------------
+
+%feature("docstring") OT::KrigingAlgorithm::setMethod
+"Linear algebra method set accessor.
+
+Parameters
+----------
+method : str
+       Used linear algebra method.
+       Value should be `LAPACK` or `HMAT`."
```
Comment (regislebrun) [inline python/src/KrigingAlgorithm_doc.i.in]: A mention of the relevant ResourceMap key would be of interest here

Comment (sofianehaddad) [reply python/src/KrigingAlgorithm_doc.i.in]: ok I will add a Note section

**Comment by sofianehaddad** [comment]
@jschueller @regislebrun I added a `reset` method to GLM. Purpose is to run again the algorithm if setter is used (noise, transformation, method). Indeed these methods are `public` and modifying internal attribut is necessary to account new elements

**Comment by jschueller** [comment]
+1

**Pull request merged**